app/src/main/java/com/yopdev/imageuploadertest/util/WSResponse.java

```diff
@@ -37,9 +44,18 @@ public void setBody(String body) {
     }
 
     public boolean isSuccess() {
-        return httpCode >= 200 && httpCode < 300;
+        return httpCode >= HTTP_OK && httpCode < HTTP_MULT_CHOICE;
     }
 
+    public boolean isOffline() {
+        return httpCode == HTTP_NOT_FOUND;
```

Comment:
At some endpoints, 404 is an expected valid response, this will cause an unexpected behaviour.

Reply:
It's perfect. The offline state does not even generate a connection tunnel and 404 does.